src/tools/perf/lib/libperf_int.h

```diff
@@ -172,13 +176,21 @@ static inline void ucx_perf_update(ucx_perf_context_t *perf,
 
     if (perf->current.time - perf->prev.time >= perf->report_interval) {
         ucx_perf_get_time(perf);
-        ucx_perf_calc_result(perf, &result);
 
 #if _OPENMP
-        /* Suppress all other threads' output */
-        #pragma omp master
-#endif /* _OPENMP */
+        /* Disable all other threads' report generation and output.
+         * The master clause cannot be used here as the unit test
+         * uct_test_perf runs on single pthreads with no parallel region,
+         * using that clause will result in undefined behavior.
+         */
+        if (omp_get_thread_num() == 0) {
```

Comment:
to avoid some code duplication, maybe write it like this:
```
#if _OPENMP
        if (omp_get_thread_num() == 0) 
#endif 
        {
            ucx_perf_calc_result(perf, &result);
            rte_call(perf, report, &result, perf->params.report_arg, 0);
        }
```

Reply:
Will do.